ols/app/models/models.py

```diff
@@ -1,9 +1,14 @@
 """Data models representing payloads for REST API calls."""
 
-from typing import Any, Optional, Self
+from typing import TYPE_CHECKING, Any, Optional, Self
 
 from pydantic import BaseModel, field_validator, model_validator
-from pydantic.dataclasses import dataclass
+
+# fixes OLS-673: Type checker reports "too many arguments" when the dataclass is constructed
+if TYPE_CHECKING:
+    from dataclasses import dataclass
+else:
+    from pydantic.dataclasses import dataclass
```

Comment:
Even though the dataclass from pydantic should have the same interface as Python's native one, I would rather ignore the error instead of importing/using something "potentially else". WDYT?

Reply:
that would be ok. There are TODOs already added there so we won't forget about those.

Reply:
yeah i'm inclined to agree with @onmete here